besu/src/main/java/org/hyperledger/besu/Runner.java

```diff
@@ -215,14 +215,16 @@ private void writeBesuPortsToFile() {
           .getLocalEnode()
           .ifPresent(
               enode -> {
-                if (enode.getDiscoveryPort().isPresent()) {
-                  properties.setProperty(
-                      "discovery", String.valueOf(enode.getDiscoveryPort().getAsInt()));
-                }
-                if (enode.getListeningPort().isPresent()) {
-                  properties.setProperty(
-                      "p2p", String.valueOf(enode.getListeningPort().getAsInt()));
-                }
+                enode
+                    .getDiscoveryPort()
+                    .ifPresent(
+                        discoveryPort ->
+                            properties.setProperty("discovery", String.valueOf(discoveryPort)));
+                enode
+                    .getDiscoveryPort()
+                    .ifPresent(
+                        listeningPort ->
+                            properties.setProperty("p2p", String.valueOf(listeningPort)));
```

Comment:
Is this change just to make it more functional style? I'd be careful doing that as it requires testing and doesn't actually improve anything.

Reply:
In this case the problem is `OptionalInt`. It's a non-generic functional-ish interface that lacks a lot of the safety features of `Optional`s so you end up having to using unsafe methods like `getAsInt`. Like I've mentioned before, using a Maybe monad is not just a matter of style. It allows the compiler to force you to handle the non-existence case explicitly, thus reducing bugs in the long run. Here's an example of a purely style change:
```
var result = 0
for (int i =0; i < arrList.size(); ++i) {
  if (i %2 ==0) result +=i
}
```
vs
```
arrList.stream().filter(i -> i %2 ==0).reduce(0, (accumulator, element) -> accumulator + element)
```
There's no additional type information here that the second has that the first doesn't have.